packages/jest-message-util/src/index.ts

```diff
@@ -145,6 +147,27 @@ export const formatExecError = (
       typeof error.stack === 'string'
         ? error.stack
         : `thrown: ${prettyFormat(error, {maxDepth: 3})}`;
+    if ('cause' in error) {
+      const prefix = '\n\nCause:\n';
+      if (typeof error.cause === 'string') {
+        cause += `${prefix}${error.cause}`;
+      } else if (typeof error.cause === 'number') {
+        cause += `${prefix}${error.cause.toString()}`;
+      } else if (error.cause instanceof Error) {
```

Comment:
use `types.isNativeError` instead - it works cross realms: https://nodejs.org/api/util.html#utiltypesisnativeerrorvalue